### PR TITLE
Minor fixes for xmap docstring, xeinsum parser

### PR DESCRIPTION
**Pull request by apaszke**
The regression loss example from the xmap docstring was broken and
the xeinsum parser didn't accept empty parens while it should.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
LGTM modulo one docstring fix.

File: jax/experimental/maps.py [hunk 1]
```diff
@@ -384,8 +384,8 @@ def xmap(fun: Callable,
     def regression_loss(x, y, w, b):
       # Contract over in_features. Batch and out_features are present in
       # both inputs and output, so they don't need to be mentioned
-      y_pred = jnp.einsum('{in_features},{in_features}->{}') + b
-      return jnp.mean((y - y_pred) ** 2, axis='batch')
+      y_pred = jnp.einsum('{in_features},{in_features}->{}', x, w) + b
+      return jnp.mean((y - y_pred) ** 2, axis=('batch', 'out_features'))
```
Comment (mattjj) [inline jax/experimental/maps.py:388]:
```suggestion
      error = jnp.sum((y - y_pred) ** 2, axis='out_features')
      return jnp.mean(error, axis='batch')
```

I think we want a sum to compute the l2 norm, then a mean over the batch, right?

Comment (apaszke) [reply jax/experimental/maps.py:388]: Right, of course! 🤦‍♂️
